core/templates/services/extension-tag-assembler.service.ts

```diff
@@ -51,29 +51,32 @@ export class ExtensionTagAssemblerService {
     const traverseSchemaAndConvertSubtitledToDicts = (
         value: Object[] | Object
     ): Object[] | Object => {
-      let result: Object[] | Object;
+      let result: Object[] | Object | undefined = undefined;
```

Comment:
```suggestion
      let result!: Object[] | Object;
```
Use non-null insertion operator(!) where you think variable is not already initialized. For reference https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1

Reply:
I did this because later on line 72 (https://github.com/YashKandalkar/oppia/blob/9f92ac8b09a891c23df7180ca3a7872d7c113b56/core/templates/services/extension-tag-assembler.service.ts#L72) 
`return result || value;` this was done, and as `result` wasn't assigned a value at first, it's `undefined` value was being used here, if `result` isn't assigned a value in the if else statements then `value` was being returned. 
Also, this if else conditions should be exhaustive correct, as the last else if is
`else if (value instanceof Object) {...}` which would be true for all types right? I am not sure, but if that's the case then we can just return `result` there instead of `result || value` and also use the non-null operator as you suggested.

Reply:
@mridul-netizen can you also take look at this

Reply:
We should lessen the use of null or undefined keeping the previous logic intact. `!` non-null assertion will be used to make the strict typing happy. The value returned will still be undefined in the case where we use non-null assertion and also suppress the error `Variable 'result' is used before being assigned.`.Try making the change suggested and check if there are still any errors.